modules/wfa-qt/src/indigo_api_callback_dut.c

```diff
@@ -1849,136 +1791,79 @@ static void append_wpas_network_default_config(struct packet_wrapper *wrapper)
 }
 #endif /* _RESERVED_ */
 
-void generate_sta_config(struct wpa_ctrl *w, struct packet_wrapper *req)
+static int configure_sta_handler(struct packet_wrapper *req, struct packet_wrapper *resp)
 {
+	char buffer[128];
 	struct tlv_hdr *tlv = NULL;
-	char buffer[64], response[16], psk[64];
-	size_t resp_len;
+	int status = TLV_VALUE_STATUS_OK, ret;
+	char *message = "DUT configured as STA";
+	char psk[64];
+
+	CHECK_RET(run_qt_command("REMOVE_NETWORK all"));
```

Comment:
For error case this would return `TLV_VALUE_STATUS_OK`, please fix that.

Reply:
This is being handled.